tests/__init.test.js

```diff
@@ -5,60 +5,67 @@
  */
 window.debug = true;
 
-navigator.xr = navigator.xr || {};
-navigator.xr.isSessionSupported = function (_sessionType) { return Promise.resolve(true); };
-navigator.xr.requestSession = function (_mode) {
-  const xrSession = new EventTarget();
-  xrSession.supportedFrameRates = [90];
-  xrSession.requestReferenceSpace = function () { return Promise.resolve(); };
-  return Promise.resolve(xrSession);
-};
-
 /* WebXR Stub */
-navigator.xr = navigator.xr || {};
+if (!navigator.xr) {
+  navigator.xr = {};
+}
 navigator.xr.isSessionSupported = function (_sessionType) { return Promise.resolve(true); };
 navigator.xr.requestSession = function (_mode) {
   const xrSession = new EventTarget();
   xrSession.supportedFrameRates = [90];
   xrSession.requestReferenceSpace = function () { return Promise.resolve(); };
+  xrSession.end = function () { return Promise.resolve(); };
   return Promise.resolve(xrSession);
 };
 
-const AFRAME = require('index');
-var AScene = require('core/scene/a-scene').AScene;
-// Make sure WebGL context is not created since CI runs headless.
-// Stubs below failed once in a while due to asynchronous test setup / teardown.
-AScene.prototype.setupRenderer = function () {};
+var AFRAME, AScene;
+var loadPromise = Promise.allSettled([
+  import('index.js'),
+  import('core/scene/a-scene.js')
+]).then(function (results) {
+  AFRAME = results[0].value.default;
+  AScene = results[1].value.AScene;
+
+  // Make sure WebGL context is not created since CI runs headless.
+  // Stubs below failed once in a while due to asynchronous test setup / teardown.
+  AScene.prototype.setupRenderer = function () {};
+});
```

Comment:
Asynchronous loading of AFRAME and a-scene, as otherwise the above WebXR stubbing would happen too late.